code-challenges.test.js

```diff
@@ -13,34 +13,188 @@
+//Pseudocode:
+// input: array
+// output: array of numbers
+// create a function divBy3 takes in an array.
+// declare a return statement run a .filter method on the array to iterate over it. Extract the numbers only by using typeof operator and setting the value strictly equal to a "number".
+// then chain a .map method to iterate through the values of the new filtered array and preform arithmetic by using the modulo operator to display the remainder only when the value is divided by 3.
+
+//
+
+const divBy3 = (array) => {
+  return array
+    .filter((value) => typeof value === "number")
+    .map((value) => value % 3)
+}
+
+// PASS  ./code-challenges.test.js
+// divBy3
+//   ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3. (2 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.422 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 1.25s.
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("sumCube", () => {
+  const cubeAndSum1 = [2, 3, 4]
+  const cubeAndSum2 = [0, 5, 10]
+  it("takes in an array of numbers and returns the sum of all the numbers cubed.", () => {
+    expect(sumCube(cubeAndSum1)).toEqual(99)
+    expect(sumCube(cubeAndSum2)).toEqual(1125)
+  })
+})
+
+// ReferenceError: sumCube is not defined
+
+// 97 |   const cubeAndSum2 = [0, 5, 10]
+// 98 |   it("takes in an array of numbers and returns the sum of all the numbers cubed.", () => {
+// >  99 |     expect(sumCube(cubeAndSum1)).toEqual(99)
+//    |     ^
+// 100 |     expect(sumCube(cubeAndSum2)).toEqual(1125)
+// 101 |   })
+// 102 | })
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+// at Object.expect (code-challenges.test.js:99:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       3 failed, 3 total
+// Snapshots:   0 total
+// Time:        3.22 s
+// Ran all test suites.
 
 // b) Create the function that makes the test pass.
+// pseudocode:
+// input:array
+// output: number
+// declare a function sumCube that takes in an array.
+// since we have to keep a running tally after the numbers are cubed declare a variable of sum equal to zero as the starting point for the count.
+// to cube each number in the array use .map method to iterate through each value in the array and preform the power of 3 arithmetic which will be stored in a declared variable of cube.
+// Using the forEach method to iterate through the cube array to add them into the sum variable up after they have been cubed.
+// will return sum that should have all the values added up.
+
+// I went back and chained the .forEach method to the cube variable. I had originally called it on the cube variable in a new line. I think I love chaining.
```

Comment:
I think I love chaining too 🤣
Nice work! You don't need the cube variable at all, since it's not actually being used anywhere!